skopt/tests/test_optimizer.py

```diff
@@ -89,3 +91,21 @@ def test_returns_result_object():
     assert isinstance(result, OptimizeResult)
     assert_equal(len(result.x_iters), len(result.func_vals))
     assert_equal(np.min(result.func_vals), result.fun)
+
+def test_acq_optimizer_generator():
+    param_list = [(ExtraTreesRegressor(random_state=2), 'lbfgs'),
+                  (RandomForestRegressor(random_state=2), 'lbfgs')]
+    for param in param_list:
+        yield check_acq_optimizer, param[0], param[1]
+
+@raises(ValueError)
```

Comment:
can we match (part of) the error message? Would make it easier for future people to know why this should raise `ValueError`.